### ISSUE TITLE
Add `SameConditionalType` config to `Style/SoleNestedConditional`

### ISSUE THREAD

**Issue by tejasbubane** [rubocop-hq/rubocop#9086]
## Is your feature request related to a problem? Please describe.

Add config to `Style/SoleNestedConditional` to check for same conditional type only. When `SameConditionalType: true` don't report offenses to merge `if` and `unless`.

## Describe the solution you'd like

current:
```ruby
# bad
if foo || !check
  unless bar && !baz
    do_something
  end
end
```
Such conditons are difficult to merge & I would like to avoid refactoring them if I don't have tests.

```ruby
# SameConditionalType: true (default false)
# bad
if foo
  if bar
    do_something
  end
end

# good
if foo
  unless bar
    do_something
  end
end
```

## Describe alternatives you've considered

Nothing

**Comment by bbatsov** [comment]
Even if I don't agree with your reasoning (for me it's all the same if I have to combine `if` and `unless`) I'm not opposed to making this configurable, although I'd probably wait a bit to see if someone else is interested in such functionality before adding it.

**Comment by bbatsov** [comment]
Closing as stale — minimal interest in 5+ years.
